omniscidb/QueryEngine/CostModel/DataSources/DwarfBench.h

```diff
@@ -21,34 +21,31 @@
 
 #include "DataSource.h"
 
-#include <bench.hpp>
-
 namespace costmodel {
 
+class DwarfBenchCoverter;
+
 // This is a temporary implementation while there is no
 // library for interaction in dwarf bench
 class DwarfBenchDataSource : public DataSource {
  public:
   DwarfBenchDataSource();
 
+  virtual ~DwarfBenchDataSource();
+
   Detail::DeviceMeasurements getMeasurements(
       const std::vector<ExecutorDeviceType>& devices,
       const std::vector<AnalyticalTemplate>& templates) override;
 
  private:
-  const size_t dwarfBenchIterations = 10;
-  const std::vector<size_t> dwarfBenchInputSizes = {256, 512, 1024, 2048};
+  const size_t dwarfBenchIterations_ = 10;
+  const std::vector<size_t> dwarfBenchInputSizes_ = {256, 512, 1024, 2048};
 
   std::vector<Detail::Measurement> measureTemplateOnDevice(ExecutorDeviceType device,
                                                            AnalyticalTemplate templ);
 
-  DwarfBench::Dwarf convertToDwarf(AnalyticalTemplate templ);
-  DwarfBench::DeviceType convertDeviceType(ExecutorDeviceType device);
-
-  std::vector<Detail::Measurement> convertMeasurement(
-      const std::vector<DwarfBench::Measurement> measurements);
-
-  DwarfBench::DwarfBench db;
+  struct PrivateImpl;
+  PrivateImpl* pimpl_;
```

Comment:
Any reason not to use a `unique_ptr`?

Reply:
I thought that because of `static_assert` in `unique_ptr` this is impossible, it turned out that it was necessary to explicitly prescribe a destructor